build/toolchain/host/BUILD.gn

```diff
@@ -42,3 +45,42 @@ gcc_toolchain("fake_${host_cpu}_gcc") {
     chip_fake_platform = true
   }
 }
+
+config("sysroot") {
+  cflags = [
+    "--sysroot=${sysroot}",
+    "--target=${target_triple}",
+  ]
+  ldflags = [
+    "--sysroot=${sysroot}",
+    "--target=${target_triple}",
+  ]
+}
+
+template("linux_clang_crossbuild_toolchain") {
+  gcc_toolchain("linux_${target_name}_clang_crossbuild") {
+    toolchain_args = {
+      current_os = "linux"
+      current_cpu = target_name
+
+      sysroot = invoker.sysroot
+      target_triple = invoker.target_triple
+      pkgconfig_libdir = "lib/${target_triple}"
+
+      is_clang = true
+      is_crossbuild = true
+
+      default_configs_extra += [ "${build_root}/toolchain/host:sysroot" ]
```

Comment:
Can you set the --sysroot option in runtime_default not adjust default_configs_extra?